api/submit_result.go

```diff
@@ -32,12 +34,14 @@ func (s *resultSubmitter) Submit(executionID string, outputKey string, outputDat
 	if err != nil {
 		return err
 	}
-	if err := exec.Complete(outputKey, outputData); err != nil {
-		return err
-	}
-	if err = s.api.execDB.Save(exec); err != nil {
-		return err
+
+	if err = exec.Complete(outputKey, outputData, rerr); err != nil {
+		exec.Error = err
+	} else {
```

Comment:
s/else/else if 

If there is only one if inside

Reply:
the mroe I think of it, the more it gets complicated, in general we should have 2 error - one for service and one for internal.

For example here if save to db fail , then it's for sure internal error because task execution is fine. The core just failed.

This error is different from the error that service return. For now, we can leave it, but not sure if it's ok

@NicolasMahe @ilgooz

Reply:
else-if seems better

Reply:
I agree, that there is two errors as one Core's internal and second is about the invalid behavior of  A Service.

To keep things simple I didn't create two types errors, actually I give it a try but code become unnecessarily complex. I'd like to experiment current implementation for now and after we can add a field like _type_ to _Error_ gRPC message type to indicate the type of error. This won't be a breaking change anyway.

Reply:
Also, we currently show only the produced error message we don't provide the inputs that are caused to error as a meta info to error. So we may also want to add field like maybe with a generic name like _data_ to keep them alongside _type_ and _message_. So client may print it to inform users. For this case _data_ can contain the _outputKey_ and _outputData_ because any of them can be sent as invalid by server.

Reply:
let's keep it simple for this PR. let's see if it useful and if it is, let's do it in another PR